src/hooks/v2/V2ContractLoader.ts

```diff
@@ -22,19 +39,21 @@ export function useV2ContractLoader() {
         // Contracts can be used read-only without a signer, but require a signer to create transactions.
         const signerOrProvider = signingProvider?.getSigner() ?? readProvider
 
-        const newContracts = Object.values(V2ContractName).reduce(
-          (accumulator, contractName) => ({
+        const contractLoaders = await Promise.all(
```

Comment:
is this just refactoring due to loadContract being async?

Reply:
yup